### PR TITLE
fix(site-stats): enforce uniqueness of wiki_site_stats per wiki, lock table when updating

### PR DESCRIPTION
**Pull request by m90**
Ticket https://phabricator.wikimedia.org/T362667

### CODE REVIEW

File: app/Jobs/UpdateWikiSiteStatsJob.php [hunk 1]
```diff
@@ -74,7 +75,8 @@ private function updateSiteStats (Wiki $wiki): void
                 $update[$field] = $value;
             }
         }
-
-        $wiki->wikiSiteStats()->updateOrCreate(['wiki_id' => $wiki->id], $update);
+        DB::transaction(function () use ($wiki, $update) {
+            $wiki->wikiSiteStats()->lockForUpdate()->updateOrCreate(['wiki_id' => $wiki->id], $update);
```
Comment (tarrow) [inline app/Jobs/UpdateWikiSiteStatsJob.php:79]: praise: Looks good; I read up a little on lockForUpdate and it look like it would do what we need it to

**Comment by m90** [comment]
Migrations apply fine locally:

```
root@api-queue-default-6ff6649d4c-9hxxl:/var/www/html# php artisan migrate

   INFO  Running migrations.

  2024_04_16_165700_enforce_site_stats_constraint ......................................................................................... 6ms DONE

root@api-queue-default-6ff6649d4c-9hxxl:/var/www/html# php artisan migrate:rollback --step 1

   INFO  Rolling back migrations.

  2024_04_16_165700_enforce_site_stats_constraint ......................................................................................... 7ms DONE

root@api-queue-default-6ff6649d4c-9hxxl:/var/www/html# php artisan migrate

   INFO  Running migrations.

  2024_04_16_165700_enforce_site_stats_constraint ......................................................................................... 6ms DONE
```

**Review by tarrow — APPROVED** [review]
looks good to me; I guess "don't forget to run the migrations" after deploying this. I wonder if we really ought to trigger these with some helm hook or something smart on future.

**Comment by m90** [comment]
> I wonder if we really ought to trigger these with some helm hook or something smart on future.

That would be fantastic.

**Pull request merged**